pkg/api/validation/validation.go

```diff
@@ -372,12 +373,35 @@ func ValidateVolumes(volumes []api.Volume, fldPath *field.Path) (sets.String, fi
 		}
 		if len(el) == 0 {
 			allNames.Insert(vol.Name)
+			if vol.VolumeSource.PersistentVolumeClaim != nil {
```

Comment:
Instead of having the value be a bool, can the value just be a ptr to the VolumeSpec?  This would make the lookup functions more generic.